toolchain/check/handle_name.cpp

```diff
@@ -235,7 +235,24 @@ auto HandleSelfTypeNameExpression(Context& context, Parse::Node parse_node)
 }
 
 auto HandleSelfValueName(Context& context, Parse::Node parse_node) -> bool {
-  return context.TODO(parse_node, "HandleSelfValueName");
+  context.node_stack().Push(parse_node);
+  return true;
+}
+
+auto HandleSelfValueNameExpression(Context& context, Parse::Node parse_node)
+    -> bool {
+  // TODO: This will find a local variable declared with name `r#self`, but
+  // should not. See #2984 and the corresponding code in
+  // HandleFunctionDefinitionStart.
+  auto name_id = context.strings().Add("self");
```

Comment:
Had you considered defining a `constexpr llvm::StringLiteral SelfValueName = "self";`, providing a canonical value to use so that typos become compile errors?

Reply:
Done.